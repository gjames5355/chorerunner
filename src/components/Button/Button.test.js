import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Button';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Button component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<MemoryRouter><Button /></MemoryRouter>, div);

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><Button /></MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Button />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})