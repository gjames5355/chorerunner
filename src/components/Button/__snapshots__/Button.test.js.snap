// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component testing renders UI as expected 1`] = `
<button
  className="Button"
/>
`;

exports[`Button component testing renders empty given no tabs 1`] = `
<button
  className="Button"
/>
`;
